src/argparse/login.rs

```diff
@@ -66,7 +62,7 @@ impl Login {
             login.delete_user(config)?;
             config.write_to_file()?;
         } else {
-            let user = login
+            let _user = login
```

Comment:
Does `login.select_user()` need to return a `User` anymore?

Reply:
I thought about it, this was definitely the easiest way to make it easily testable I think

Reply:
Since we don't use the result in our code, I'm not sure we need to have this externally visible. Surely we can already test that `select_user()` is behaving correctly with `MockConfigManager`?

Reply:
How do we assert on the value about the user from that?

Reply:
You can use `.with()` and `.withf()`. Take a look at `pexshell::cli::login::tests::test_select_user_add_no_verify` for examples of both. You should use the more declarative `with()` syntax over `withf()` where possible, but you can always fall back on `withf()` when you need more flexibility.

Reply:
But that will only check a function is called with the right things, how would I assert that the last time and current user is set correctly

Reply:
You could check the config file has the correct values in, although tbh I would just unit-test the hell out of `Config` to cover that case.

Reply:
Different piece of work, unittesting that I think.